packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -93,13 +94,17 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
   }
 
   debug('Initiating execution of changeset %s on stack %s', changeSetName, deployName);
-  await cfn.executeChangeSet({ StackName: deployName, ChangeSetName: changeSetName }).promise();
-  // tslint:disable-next-line:max-line-length
-  const monitor = options.quiet ? undefined : new StackActivityMonitor(cfn, deployName, options.stack, (changeSetDescription.Changes || []).length).start();
-  debug('Execution of changeset %s on stack %s has started; waiting for the update to complete...', changeSetName, deployName);
-  await waitForStack(cfn, deployName);
-  if (monitor) { await monitor.stop(); }
-  debug('Stack %s has completed updating', deployName);
+  if (options.execute) {
+    await cfn.executeChangeSet({StackName: deployName, ChangeSetName: changeSetName}).promise();
+    // tslint:disable-next-line:max-line-length
+    const monitor = options.quiet ? undefined : new StackActivityMonitor(cfn, deployName, options.stack, (changeSetDescription.Changes || []).length).start();
+    debug('Execution of changeset %s on stack %s has started; waiting for the update to complete...', changeSetName, deployName);
+    await waitForStack(cfn, deployName);
+    if (monitor) {
+      await monitor.stop();
+    }
+    debug('Stack %s has completed updating', deployName);
+  }
   return { noOp: false, outputs: await getStackOutputs(cfn, deployName), stackArn: changeSet.StackId! };
```

Comment:
clarification: just verifying that this would not count as a noOp as we are still generating a ChangeSet and mutating the state of the stack.

Reply:
I modified this portion of code but it doesn't show as outdated. 
I tried to not interfere with the testing scripts for as much as I could. The executeChangeSet method is needed to be called in `deploy-stacks.ts` in order for some tests to pass. I think calling `cfn.executeChangeSet()`  without the promise class. This way the tests will pass successfully without modifying them.

Reply:
I think it's a good call not to modify existing tests so we don't mask a potential regression.

Reply:
I disagree. The fact that tests fail is an indication of a regression, and indeed we've had a regression here where bootstrap stacks failed to execute because "execute" was not passed to it.